mlflow/utils/requirements_utils.py

```diff
@@ -409,3 +414,36 @@ def _get_pinned_requirement(package, version=None, module=None):
             version = version_raw
 
     return f"{package}=={version}"
+
+
+def _convert_package_name_to_module_name(package_name):
+    _init_modules_to_packages_map()
+
+    package_name = _normalize_package_name(package_name)
+
+    for module, pkg_list in _MODULES_TO_PACKAGES.items():
+        if package_name in pkg_list:
+            return module
+
+    return package_name
+
+
+def _check_pkg_installed_version_satisfy_requirements(req_line):
+    req = pkg_resources.Requirement.parse(req_line)
+    pkg_name = req.name
+    module = _convert_package_name_to_module_name(pkg_name)
+
+    try:
+        installed_version = _get_installed_version(pkg_name, module)
+    except ModuleNotFoundError:
+        return f"package {pkg_name} is not installed but {req_line} required"
+
+    if len(req.specifier) == 0:
+        return None
+
+    if req.specifier.contains(installed_version):
```

Comment:
This method will verify `installed_version` satisfy all specifiers . The req.specifier is a `SpecifierSet` instance.